src/support/rand.c

```diff
@@ -56,19 +56,23 @@
 #define DEFAULT_SEED_W 521288629
 #define DEFAULT_SEED_Z 362436069
 
+/* Circular shift is converted by compilers to a "rol"-like CPU instruction. */
+#define WT_LEFT_CIRCULAR_SHIFT32(x, nbits) (((x) << (nbits)) | ((x) >> (32 - (nbits))))
+
+/* Make a seed from session id, time (in nanoseconds) and pid. */
+#define MAKE_SEED(id, t, pid)                                                           \
+    (((uint64_t)WT_LEFT_CIRCULAR_SHIFT32((uint64_t)(id) + 1, 3)) + ((t) / WT_BILLION) + \
```

Comment:
Is it possible for a MongoDB to open/close the same session very fast using in the same process? Worth testing at least.

Reply:
This is not an issue since a session with one id is only initialized once in the lifetime of application.

Reply:
Actually it may be possible, When a session is closed it can be still alive in WT for re-use. If MongoDB chooses to open session and happens to get the same session we can run into this situation.